File: 2756-buy-two-chocolates/buy-two-chocolates.py
'''
just find the 2 min priced choclate and check if you can buy it , successfully with leftove money , 
if yes return remaining money , else dont buy and return what money you have already 

t  n
s  1 

'''



#solution 1
class Solution:
    def buyChoco(self, prices: List[int], money: int) -> int:
        

        min_1,min_2 = float(inf), float("inf")
        n = len(prices)

        for i in range(n):

            if prices[i]<=min_1:
                min_2  = min_1
                min_1 = prices[i]
            elif prices[i]<min_2:
                min_2 = prices[i]
            
        left = money-(min_1+min_2)
        if left<0:
            return money
        return left

File: 2756-buy-two-chocolates/buy-two-chocolates.java
class Solution {
    public int buyChoco(int[] prices, int money) {
        int min_1 = Integer.MAX_VALUE;
        int min_2 = Integer.MAX_VALUE;
        int n = prices.length;

        for (int i = 0; i < n; i++) {
            if (prices[i] <= min_1) {
                min_2 = min_1;
                min_1 = prices[i];
            } else if (prices[i] < min_2) {
                min_2 = prices[i];
            }
        }

        int left = money - (min_1 + min_2);
        if (left < 0) {
            return money;
        }
        return left;
    }
}